### PR TITLE
Added login page and authentication checks

### PR DESCRIPTION
**Pull request by kentkwu**
Sorry I merged the other one. I messed up git on my end and didn't want to have to stack these pull requests. Just wanted to keep moving forward
- Added the login page and some checks for authentication
- Added actions for navigating with router. This makes it easier to debug because you can track actions with devtools
  screenshot 2016-08-23 18 08 20

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,20 +1,46 @@
...
+import Dashboard from './components/dashboard/Dashboard.react.js'
+import { loginSuccess } from './actions/LoginActions.js'
 
 injectTapEventPlugin()
 
+const getRoutes = (reduxStore: Object) => {
+  const authRequired = (nextState: Function, replace: Function) => {
+    const state = reduxStore.getState()
+
+    if (localStorage.authToken) {
```
Comment (dsiah) [inline src/App.js:20]: create a TODO for authenticating this token please

File: src/actions/LoginActions.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+// @flow
+import request from 'superagent'
```
Comment (dsiah) [inline src/actions/LoginActions.js:2]: which should we use superagent or the native node request library? any argument for superagent?

File: src/components/signup/SignupForm.react.js [hunk 3]
```diff
@@ -37,7 +36,7 @@ const SignupForm = (props: Object) => {
   /* eslint-enable */
 
   return (
-    <Paper zDepth={1} className="signupForm">
+    <Paper zDepth={1} className="form">
```
Comment (dsiah) [inline src/components/signup/SignupForm.react.js:39]: Add a button and action to switch from signup to login (ex. I accidentally click signup and need to get back to login form)

Maybe leave this as an exercise for me to do this to get use to our workflow

File: src/reducers/Reducers.js [hunk 4]
```diff
@@ -3,17 +3,37 @@ import { combineReducers } from 'redux'
 import { reducer as formReducer } from 'redux-form'
 import LoginConstants from '../constants/LoginConstants.js'
 
-const initialState = {
+const appInitialState = {
 }
 
-const appState = (state: Object = initialState, action: Object) => {
+const loginInitialState = {
+  authToken: null,
```
Comment (dsiah) [inline src/reducers/Reducers.js:10]: what are we planning on using for authTokens?

**Comment by dsiah** [comment]
Ship: Just clarify some of my line comments. And can you provide a link with instructions for the dev tools -- " Added actions for navigating with router. This makes it easier to debug because you can track actions with devtools "

File: src/reducers/Reducers.js (hunk 4 above)
Comment (kentkwu) [reply src/reducers/Reducers.js:10]: We already have the backend set up. It uses the same thing (JWT) as it did last semester

File: src/components/signup/SignupForm.react.js (hunk 3 above)
Comment (kentkwu) [reply src/components/signup/SignupForm.react.js:39]: Yeah that's a good idea. Open up an issue?

File: src/actions/LoginActions.js (hunk 2 above)
Comment (kentkwu) [reply src/actions/LoginActions.js:2]: I ran into some issues with request and webpack/babel that I couldn't manage to fix. That's why I switched to superagent

File: src/App.js (hunk 1 above)
Comment (kentkwu) [reply src/App.js:20]: can you clarify?

**Pull request merged**